File: mvp/src/main/java/com/thefuntasty/taste/TasteActivity.java
package com.thefuntasty.taste;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;

public abstract class TasteActivity extends AppCompatActivity {

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		int activityLayout = getActivityLayout();
		if (activityLayout != 0) {
			setContentView(activityLayout);
			ButterKnife.bind(this);
		}
		inject();
	}

	public abstract @LayoutRes int getActivityLayout();

	public abstract void inject();

}

File: mvp/src/main/java/com/thefuntasty/taste/TasteActivity.kt
package com.thefuntasty.taste

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import butterknife.ButterKnife

abstract class TasteActivity : AppCompatActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val activityLayout = activityLayout
		if (activityLayout != 0) {
			setContentView(activityLayout)
			ButterKnife.bind(this)
		}
		inject()
	}

	abstract val activityLayout: Int

	abstract fun inject()

}
